CIS-3760-Team-25/unity-checkers
# Player Select Screen

**Description**

Adds Player Select screen and a temporary Main Menu to the GameScene. Also added class for API interaction 


**Testing**

Manually test overlay transitions
 - Click play button from Main Menu, ensure Player Select screen is displayed
 - Click help button from Main Menu, ensure nothing happens (screen not implemented yet)
 - Click stats button from Main Menu, ensure nothing happens (screen not implemented yet)
 - Click Play button from Player Select screen, ensure game starts

Manually test player input validation
 - Test trying to start game with no player info given (game can't be started, errors show)
 - Test trying to start game with 1 player as guest (game can't be started, errors show)
 - Test trying to start game with both players as guest (game started, game record created)
 - Test trying to start game with both player providing name/email (game started, game record created)


**Related Issues**

Closes #47 


**Definition of Done**

Before Merging

- [x] Code conforms to coding standards (formatted/linted)
- [x] Code is properly documented (comments, model updated)
- [x] Code has been tested and the testing method is documented in this PR
- [x] Code doesn't introduce any new bugs
- [x] Code is reviewed by at least 2 people
- [x] Kanban board is updated

After Merging

- [ ] GitHub branches related to the story are deleted
- [ ] Associated GitHub issue is updated/closed
- [ ] Kanban board is updated
